pallets/liquidity-pools/src/lib.rs

```diff
@@ -693,20 +697,18 @@ pub mod pallet {
 		#[pallet::weight(10_000)]
 		#[pallet::call_index(10)]
 		pub fn schedule_upgrade(origin: OriginFor<T>, contract: DomainAddress) -> DispatchResult {
-			match contract {
-				DomainAddress::EVM(_, _) => {
-					let who = ensure_signed(origin)?;
-
-					T::OutboundQueue::submit(
-						who,
-						contract.domain(),
-						Message::ScheduleUpgrade {
-							contract: contract.address(),
-						},
-					)
-				}
-				_ => Err(DispatchError::Other("Not an EVM domain address")),
-			}
+			ensure_root(origin)?;
+			// get account id of soverign account
```

Comment:
```suggestion
			// get account id of this pallet
```

Reply:
@offerijns with that said, this would mean that this pallet itself would be paying for the fees for executing this call. So we would need to first transfer sufficient CFG to the account derived from the pallet id before we could call this extrinsic

Reply:
cc @mustermeiszer

Reply:
you all forget what I said above, this extrinsic is called by the root origin, I was mixing stuff up. All good!